### PR TITLE
Nostr:  Auto generate nostr secret 

### PR DESCRIPTION
**Pull request by abhay-raizada**
1. The nostr secret is auto-generated on mount in homescreen
2. Reveal nostr secret is added to the settings screen: 
image

3. You should be able to copy your nostr secret to clipboard:
image

### CODE REVIEW

File: app/hooks/use-nostr-profile.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        console.log("Generated new nostr secret:", nostrSecret)
+        await Keychain.setInternetCredentials(
+          KEYCHAIN_NOSTRCREDS_KEY,
+          KEYCHAIN_NOSTRCREDS_KEY,
+          nostrSecret,
+        )
+        setNostrSecretKey(nostrSecret)
+        return nostrSecret
+      } catch (error) {
+        console.error("Error in getMnemonic: ", error)
```
Comment (abhay-raizada) [inline app/hooks/use-nostr-profile.ts:29]: error in generating nostr secret*

File: app/screens/settings-screen/show-nostr-secret.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      <View style={{ backgroundColor: "white" }}>
+        <Text type="h2">Your nostr address is: {"\n"}</Text>
+        <Text>
+          {nostrSecretKey} {"\n"}
+        </Text>
+        <Text onPress={copyToClipboard}>{copied ? "Copied!" : "Click to copy"}</Text>
+      </View>
+    </ReactNativeModal>
+  )
+}
```
Comment (Nodirbek75) [inline app/screens/settings-screen/show-nostr-secret.tsx:42]: Screenshot_1708022416

Can you make better design for this modal, please?

**Review by Nodirbek75 — CHANGES_REQUESTED** [review]

**Comment by Nodirbek75** [comment]
@abhay-raizada why did you fork the project and making the changes here. You could make different branch and work in original repo? It would be easier for me also to run the app and check the changes.

**Comment by islandbitcoin** [comment]
> @abhay-raizada why did you fork the project and making the changes here. You could make different branch and work in original repo? It would be easier for me also to run the app and check the changes.

@abhay-raizada I just added you to the GitHub repo, you should now be able to make branches.

File: app/screens/settings-screen/show-nostr-secret.tsx (hunk 2 above)
Comment (islandbitcoin) [reply app/screens/settings-screen/show-nostr-secret.tsx:42]: @abhay-raizada we have been using https://bitcoin.design as a guide for many of the screens, you might find some good templates there.

Comment (abhay-raizada) [reply app/screens/settings-screen/show-nostr-secret.tsx:42]: Sure, on it

**Comment by abhay-raizada** [comment]
@islandbitcoin  tried to push my branch, i'm still getting a permission denied error so continuing working on the fork

File: app/screens/settings-screen/show-nostr-secret.tsx (hunk 2 above)
Comment (abhay-raizada) [reply app/screens/settings-screen/show-nostr-secret.tsx:107]: done

**Comment by islandbitcoin** [comment]
> @islandbitcoin tried to push my branch, i'm still getting a permission denied error so continuing working on the fork

Abhay I changed your permissions on the repo
can you try making a branch again?
you will need to accept the invite in your email

**Comment by abhay-raizada** [comment]
@islandbitcoin  done! created this PR: https://github.com/lnflash/flash-mobile/pull/91 , closing this one in favor of that.

**Pull request closed without merging**